timur/lib/client/jsx/components/search/search.jsx

```diff
@@ -2,138 +2,330 @@ import Pager from '../pager';
 import React, {Component} from 'react';
 import {connect} from 'react-redux';
 
+import * as _ from 'lodash';
+
+import {css} from '@emotion/core';
+import GridLoader from 'react-spinners/GridLoader';
+
 import SelectInput from '../inputs/select_input';
-import { selectModelNames } from '../../selectors/magma';
-import { requestTSV, requestModels, requestDocuments } from '../../actions/magma_actions';
-import { selectSearchCache } from '../../selectors/search_cache';
-import { cacheSearchPage, setSearchPageSize, setSearchPage, emptySearchCache } from '../../actions/search_actions';
+import {
+  selectModelNames,
+  selectTemplate,
+  displayAttributes
+} from '../../selectors/magma';
+import {
+  requestTSV,
+  requestModels,
+  requestDocuments
+} from '../../actions/magma_actions';
+import {
+  selectSearchCache,
+  selectSearchAttributeNames
+} from '../../selectors/search_cache';
+import {
+  cacheSearchPage,
+  setSearchPageSize,
+  setSearchPage,
+  emptySearchCache,
+  setSearchAttributeNames
+} from '../../actions/search_actions';
 
 import ModelViewer from '../model_viewer';
+import TreeView, {getSelectedLeaves} from 'etna-js/components/TreeView';
+
+const attributeOptions = [
+  [
+    'Top Level 1',
+    [
+      ['Inner 1'],
+      ['Inner 2', [['Inner Inner 1'], ['Inner Inner 2'], ['Inner Inner 3']]],
+      ['Inner 3']
+    ]
+  ],
+  ['Top Level 2']
+];
+
+const spinnerCss = css`
+  display: block;
+  margin: 2rem auto;
+`;
 
-class Search extends Component {
+export class Search extends Component {
```

Comment:
Export the raw component (unwrapped with `redux` `connect`), to make some tests a little bit easier to do.